File: cpp/3254.find-the-power-of-k-size-subarrays-i.cpp
/*
 * @lc app=leetcode id=3254 lang=cpp
 *
 * [3254] Find the Power of K-Size Subarrays I
 */

// @lc code=start
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) {
        std::vector<int> results;
        int n = nums.size();

        for (int i = 0; i <= n - k; ++i) {
            bool isConsecutive = true;
            for (int j = 0; j < k - 1; ++j) {
                if (nums[i + j] + 1 != nums[i + j + 1]) {
                    isConsecutive = false;
                    break;
                }
            }
            if (isConsecutive) {
                results.push_back(*std::max_element(nums.begin() + i, nums.begin() + i + k));
            } else {
                results.push_back(-1);
            }
        }

        return results;
    }
};
// @lc code=end

File: scala/3254.find-the-power-of-k-size-subarrays-i.scala
/*
 * @lc app=leetcode id=3254 lang=scala
 *
 * [3254] Find the Power of K-Size Subarrays I
 */

// @lc code=start
object Solution {
    def resultsArray(nums: Array[Int], k: Int): Array[Int] = {
        nums.sliding(k).map { subarr =>
            if (subarr.zip(subarr.tail).forall {
                case (a, b) => b == a + 1
            }) {
                subarr.max
            } else {
                -1
            }
        }.toArray
    }
}
// @lc code=end

